Modules/LearningSequence/classes/Player/class.ilObjLearningSequenceLPPollingGUI.php

```diff
@@ -0,0 +1,43 @@
+<?php declare(strict_types=1);
+
+class ilObjLearningSequenceLPPollingGUI
+{
+    const PARAM_LSO_LP_OBJID = LSUrlBuilder::PARAM_LSO_PARAMETER;
+
+    protected ilCtrl $ctrl;
+    protected int $current_user_id;
+    protected ilObjectDataCache $obj_data_cache;
+
+    public function __construct(
+        ilCtrl $ctrl,
+        int $current_user_id,
+        ilObjectDataCache $obj_data_cache
+    ) {
+        $this->ctrl = $ctrl;
+        $this->current_user_id = $current_user_id;
+        $this->obj_data_cache = $obj_data_cache;
+    }
+
+    public function executeCommand()
+    {
+        $cmd = $this->ctrl->getCmd();
+        switch ($cmd) {
+            case LSControlBuilder::CMD_CHECK_CURRENT_ITEM_LP:
+                $this->getCurrentItemLearningProgress();
+                // no break
+            default:
+                throw new ilException("Command not supported: $cmd");
+        }
+    }
+    
+    protected function getCurrentItemLearningProgress()
+    {
+        $obj_id = (int) $_GET[self::PARAM_LSO_LP_OBJID];
```

Comment:
Please do not use $_GET but `retrieve` from the new HTTP wrapper instead. @daniwe4 may give hints.